### PR TITLE
Add Dockerfile.release for bpftrace docker image on quay.io

### PR DESCRIPTION
**Pull request by dalehamel**
For #1051 this sets up a release docker image.

I'm testing this out on my own fork to make sure it pushes as expected before I mark this as ready to review. Locally it produces a ~100MB image with a statically linked (+libc) bpftrace.

I've configured my own quay.io/dalehamel/bpftrace repo to use the same triggers as iovisor/bpftrace and am running a build now.

### CODE REVIEW

**Comment by dalehamel** [comment]
bad news, the default build timeout on quay.io is 2 hours, which isn't enough time to build LLVM/Clang as "MinSizeRel", rather than the distro default "RelWithDebInfo". By building only libclang, I can get a 76MB bpftrace exe, but this is more than twice the size of the former.

@4ast it looks like you have access to both quay.io and bpftrace repo settings, are you able to add the secret for the bot account from https://quay.io/organization/iovisor?tab=robots for "bpftrace_buildbot" as per https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets 

This way we can do something similar to what kubect-trace does on travis with encrypted secrets, but with github actions (https://github.com/iovisor/kubectl-trace/blob/master/hack/ci-release-image.sh#L15-L18). If the bot secret were added as "QUAY_TOKEN" then it should be easy to use this same approach.

We already build this artifact on github actions, so it should be almost a no-op to do this build there, but we'll need that secret.

cc @danobi

**Comment by dalehamel** [comment]
>  the default build timeout on quay.io is 2 hours,

It looks like quay.io jobs might actually eventually complete, I do see one successful job and it is able to build and push the smaller image.

We can still merge this if that's the case, it'll just be a as jobs are retried. I think that the QUAY_TOKEN is probably more reliable though, and would still suggest we go this route if possible.

**Comment by dalehamel** [comment]
https://quay.io/repository/dalehamel/bpftrace/build/938bf94d-aaca-455f-8175-af605d6dddd6 here is the docker image build, the resulting layer is 40.6 MB 

The docker pull is pretty fast, and I can run the image with

```
docker run quay.io/dalehamel/bpftrace:quay-release bpftrace
```

**Comment by dalehamel** [comment]
@danobi even though the build is slow, I think we should still merge. I can follow up about the QUAY_TOKEN thing on #1051

File: docker/Dockerfile.release [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# This dockerfile is used for quay.io images
+FROM ubuntu:bionic as builder
+
+ARG bcc_ref="v0.12.0"
+ARG LLVM_VERSION="8"
+ENV LLVM_VERSION=$LLVM_VERSION
+
+# There is no need need for most of this once
+# https://github.com/iovisor/bcc/pull/2677 merges, as this is all to build
```
Comment (fbs) [inline docker/Dockerfile.release]: https://github.com/iovisor/bcc/pull/2677 looks merged

Comment (dalehamel) [reply docker/Dockerfile.release]: Sorry comment should say “Released”. The packages on the iovisor apt repo are also out of date. Once bcc 0.13.0 is out we shouldn’t need to build it to get all of the static libs, but for now we still do.

I’ll update the comment.

**Comment by dalehamel** [comment]
I added the /tools directory at /usr/local/bin since the size difference is minimal. Now out of the box you can do things like:

```
docker run -ti --privileged -v /sys/kernel/debug:/sys/kernel/debug:rw -v /lib/modules/:/lib/modules:ro  -v /usr/src:/usr/src:ro quay.io/iovisor/bpftrace tcpconnect.bt
```

Which is pretty cool I think. @fbs this should be good to merge

**Comment by dalehamel** [comment]
thanks @danobi I'll follow up with another PR to use the token + github actions. If that works as I expect we can revert this, but good to have it pushing tags / branches now (even if it is slow)

**Pull request merged**